test/powershell/Modules/Microsoft.PowerShell.Management/Set-Content.Tests.ps1

```diff
@@ -1,51 +1,106 @@
-      $AsItWas |Should BeExactly "ExpectedContent"
-      {set-content -path testdrive:\$file1 -value $null -ea stop} | Should Not Throw
-      $AsItIs=get-content testdrive:\$file1
-      $AsItIs| Should Not Be $AsItWas
-    }
-    It "should throw 'ParameterArgumentValidationErrorNullNotAllowed' when -Path is `$null" {
-      Try {set-content -path $null -value "ShouldNotWorkBecausePathIsNull" -ea stop; Throw "Previous statement unexpectedly succeeded..."
-      } Catch {$_.FullyQualifiedErrorId | Should Be "ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SetContentCommand"}
-    }
-    #[BugId(BugDatabase.WindowsOutOfBandReleases, 903880)]
-    It "should throw 'ParameterArgumentValidationErrorNullNotAllowed' when -Path is `$()" {
-      Try {set-content -path $() -value "ShouldNotWorkBecausePathIsInvalid" -ea stop; Throw "Previous statement unexpectedly succeeded..."
-      } Catch {$_.FullyQualifiedErrorId | Should Be "ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SetContentCommand"}
-    }
-    #[BugId(BugDatabase.WindowsOutOfBandReleases, 906022)]
-    It "should throw 'PSNotSupportedException' when you set-content to an unsupported provider" -Skip:($IsLinux -Or $IsOSX) {
-      Try {set-content -path HKLM:\\software\\microsoft -value "ShouldNotWorkBecausePathIsUnsupported" -ea stop; Throw "Previous statement unexpectedly succeeded..."
-      } Catch {$_.FullyQualifiedErrorId | Should Be "NotSupported,Microsoft.PowerShell.Commands.SetContentCommand"}
+}
+
+Describe "Set-Content should work for PSDrive with UNC path as root" -Tags @('CI', 'RequireAdminOnWindows') {
+    BeforeAll {
+        $file1 = "file1.txt"
+        $filePath1 = join-path $testdrive $file1
+        #create a random folder 
+        $randomFolderName = "TestFolder_" + (Get-Random).ToString()
+        $randomFolderPath = join-path $testdrive $randomFolderName
+        $null = New-Item -Path $randomFolderPath -ItemType Directory -ErrorAction SilentlyContinue
     }
-    #[BugId(BugDatabase.WindowsOutOfBandReleases, 9058182)]
-    It "should be able to pass multiple [string]`$objects to Set-Content through the pipeline to output a dynamic Path file" -Pending:($IsLinux -Or $IsOSX) {#https://github.com/PowerShell/PowerShell/issues/891
-      "hello","world"|set-content testdrive:\dynamicfile2.txt
-      $result=get-content testdrive:\dynamicfile2.txt
-      $result.length |Should be 2
-      $result[0]     |Should be "hello"
-      $result[1]     |Should be "world"
+    It "should create a file in a psdrive with UNC path as root" -skip:(-not $IsWindows){
+        try
+        {
+            # create share
+            net share testshare=$randomFolderPath /grant:everyone,FULL
+            New-PSDrive -Name Foo -Root \\localhost\testshare -PSProvider FileSystem
+            $result=set-content -path Foo:\$file1 -value "$file1" -passthru
+            $result| Should be "$file1"
```

Comment:
I'm not sure whether `set-content -passthru` actually retrieves the data from the file - would an explicit call to `get-content` be better?